core/services/gateway/config/config.go

```diff
@@ -19,6 +19,7 @@ type ConnectionManagerConfig struct {
 }
 
 type DONConfig struct {
+	ChainId       string
```

Comment:
This doesn't seem to be relevant to every DON. What if there's a DON that works with multiple chains, like CCIP?  What if somebody wants to check data on multiple chains in their handler? In Functions we're using it specifically for the allowlist so it makes sense to put it in Handler's config.

Reply:
Convinced. Then I think we can take the chain ID from the job spec. I want to avoid specifying another Chain ID just for Handler, because a simple copy-paste error will make all parts inconsistent.

Reply:
Updated.